### ISSUE TITLE
Is unixodbc-dev necessary on macOS when build?

### ISSUE THREAD

**Issue by arielyang** [taosdata/TDengine#12053]
**Bug Description**
I build TDengine on macOS, and that indicates unixodbc-dev should be installed.

**To Reproduce**

Environment: MacBook Pro M1 Pro, macOS
Step: execute "cmake .. && cmake --build ."

I got message below:
```
CMake Warning at src/connector/odbc/CMakeLists.txt:66 (message):
unixodbc is not installed yet, you may install it under ubuntu by typing:
brew install unixodbc
```
Then I installed unixodbc, and execute cmake again, I got message below:
```
CMake Warning at src/connector/odbc/CMakeLists.txt:47 (message):
unixodbc-dev is not installed yet, you may install it with homebrew by
typing: brew install unixodbc

```
Maybe unixodbc should be unixodbc-dev here?

Finally, I build successfully without unixodbc-dev.

Should "unixodbc" be install on macOS？

**Expected Behavior**

Correct output message and ensure if it is necessary of unixodbc-dev.

**Screenshots**
N/A.

**Environment (please complete the following information):**
 - OS: MacBook Pro M1 Pro, macOS
 - TDengine Version: latest source code

**Additional Context**
N/A.

**Comment by arielyang** [comment]
@freemine please check this

**Comment by arielyang** [comment]
@sangshuduo already fix some typo in #12047

**Comment by freemine** [comment]
see PR in #12068
